playbooks/common/openshift-cluster/evaluate_groups.yml

```diff
@@ -8,7 +8,7 @@
   - name: Evaluate groups - g_etcd_hosts or g_new_etcd_hosts required
     fail:
       msg: This playbook requires g_etcd_hosts or g_new_etcd_hosts to be set
-    when: "{{ g_etcd_hosts is not defined and g_new_etcd_hosts is not defined}}"
+    when: g_etcd_hosts is not defined or g_new_etcd_hosts is not defined
```

Comment:
Seems like there's a logic change here?

Reply:
This was intentional because the failure message said `or`, and it made sense to be `or` instead of `and`.  I can flip it back if this difference is intentional.  Either the msg needs to be updated or the when condition.

Reply:
I think the message is equivalent to the logic check. If A or B needs to be defined then the check would be to fail if A and B are undefined, right? Just realized i merged other branches ahead of this one so we'll want to align them.

Reply:
If the logic should actually be `and`, we should make this consistent with the next two tasks as well. 
```yaml
   - name: Evaluate groups - g_master_hosts or g_new_master_hosts required
      fail:
       msg: This playbook requires g_master_hosts or g_new_master_hosts to be set
     when: g_master_hosts is not defined or g_new_master_hosts is not defined
 
   - name: Evaluate groups - g_node_hosts or g_new_node_hosts required
     fail:
       msg: This playbook requires g_node_hosts or g_new_node_hosts to be set
     when: g_node_hosts is not defined or g_new_node_hosts is not defined
```

Reply:
@abutcher can you vet these conditions? for new master, new node, new etcd?

Reply:
It reads "I need a or b, if you have neither, fail."  I think the current or in when statement is probably incorrect.

Reply:
I've removed the conditional change from this PR and will open a new PR to address all related tasks.